### PR TITLE
Prepopulate config computed values after sdk initial population

### LINKED ISSUE #3362: Launching Questionnaire with unique ID configs causes app to crash
**Describe the bug**
Launching a Questionnaire with unique ID assignment configs causes the app to crash.
This happens for 2 reasons:
1. The data query that filters Group resources by `managing-entity` requires a value provided by a computed rule. This requires checking the computedValuesMap. Since the map isn't passed to the filter implementation, retrieving that value throws an exception
2. When launching the Questionnaire, the extracted ID is set as an initial value in the field (QuestionnaireItem). This happens in [retrieveQuestionnaire](https://github.com/opensrp/fhircore/blob/main/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt#L242). Subsequently in [viewModel.populateQuestionnaire(questionnaire, this.questionnaireConfig, actionParameters)](https://github.com/opensrp/fhircore/blob/main/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt#L280) which calls ->  [ResourceMapper.populate(questionnaire, launchContexts)](https://github.com/opensrp/fhircore/blob/main/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt#L1094), leads to the Questionnaire item having initial value and initial expression which isn't allowed by the SDC and an exception is thrown

**To Reproduce**
Steps to reproduce the behaviour:
1. Add unique ID configs for a Questionnaire
2. Launch the Questionnaire

**Expected behaviour**
The questionnaire should launch and the ID field (e.g. PHN) should have the unique ID prepopulated
If editing the registration info, the assigned ID should be displayed in the ID field. 

**Additional context**
*App stacktrace*
`Process: org.smartregister.opensrp.diabetesCompassClinic, PID: 10256
                                                                                                    java.lang.IllegalStateException: QuestionnaireItem item is not allowed to have both initial.value and initial expression. See rule at http://build.fhir.org/ig/HL7/sdc/expressions.html#initialExpression.
                                                                                                    	at com.google.android.fhir.datacapture.mapping.ResourceMapper.populateInitialValue(ResourceMapper.kt:247)
                                                                                                    	at com.google.android.fhir.datacapture.mapping.ResourceMapper.populateInitialValues(ResourceMapper.kt:240)
                                                                                                    	at com.google.android.fhir.datacapture.mapping.ResourceMapper.populate(ResourceMapper.kt:230)
                                                                                                    	at org.smartregister.fhircore.quest.ui.questionnaire.QuestionnaireViewModel.populateQuestionnaire(QuestionnaireViewModel.kt:1095)
                                                                                                    	at org.smartregister.fhircore.quest.ui.questionnaire.QuestionnaireActivity.buildQuestionnaireFragment(QuestionnaireActivity.kt:280)
                                                                                                    	at org.smartregister.fhircore.quest.ui.questionnaire.QuestionnaireActivity.access$buildQuestionnaireFragment(QuestionnaireActivity.kt:63)
                                                                                                    	at org.smartregister.fhircore.quest.ui.questionnaire.QuestionnaireActivity$renderQuestionnaire$1.invokeSuspend(QuestionnaireActivity.kt:246)`

### PR DESCRIPTION
**Pull request by LZRS**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #3362 

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt [hunk 1]
```diff
@@ -140,94 +140,99 @@ constructor(
...
     if (questionnaireConfig.id.isEmpty() || questionnaireConfig.id.isBlank()) return null
+    return defaultRepository.loadResource<Questionnaire>(questionnaireConfig.id)
+  }
+
+  /**
+   * Pre-populates questionnaire with computed values from the Rules engine as well as include
+   * initial values set on configured [QuestionnaireConfig.barcodeLinkId] or
+   * [QuestionnaireConfig.uniqueIdAssignment] properties.
+   */
+  private suspend fun Questionnaire.prepopulateWithComputeConfigValues(
```
Comment (allan-on) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:157]: Based on the comment, would it be better if we have this method named **prepopulateWithComputedConfigValues** ?

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt [hunk 2]
```diff
@@ -140,94 +140,99 @@ constructor(
...
+    config: QuestionnaireConfig,
+    actionParameters: List<ActionParameter>?,
+  ) {
+    require(config.id.isNotBlank())
 
     // Compute questionnaire config rules and add extra questionnaire params to action parameters
     val questionnaireComputedValues =
-      questionnaireConfig.configRules?.let {
+      config.configRules?.let {
```
Comment (allan-on) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:165]: Is there a reason we no longer want the config variable explicitly named as it was before?

**Review by allan-on — COMMENTED** [review]
Looks good. Works as expected. See comments on readability

**Review by ellykits — DISMISSED** [review]

**Comment by ellykits** [comment]
Resolve the failing tests to merge.

**Review by ellykits — APPROVED** [review]

**Review by allan-on — APPROVED** [review]

**Review by ellykits — APPROVED** [review]

**Pull request merged**